client/my-sites/plugins-wpcom/jetpack-plugins-panel.jsx

```diff
@@ -0,0 +1,220 @@
+		} );
+	}
+
+	onNavSearch = searchTerm => {
+		this.setState( {
+			searchTerm,
+		} );
+	}
+
+	getSearchPlaceholder() {
+		const { translate } = this.props;
+		switch ( this.state.selectedGroup ) {
+			case 'all':
+				return translate( 'Search All…', { textOnly: true } );
+			case 'engagement':
+				return translate( 'Search Engagement…', { textOnly: true } );
+			case 'security':
+				return translate( 'Search Security…', { textOnly: true } );
+			case 'appearance':
+				return translate( 'Search Appearance…', { textOnly: true } );
+			case 'writing':
+				return translate( 'Search Writing…', { textOnly: true } );
+		}
+	}
+
+	getSelectedText() {
+		const found = find( this.getNavItems(), item => this.state.selectedGroup === item.key );
+		return found ? found.title : '';
+	}
+
+	getNavItems() {
+		const { translate } = this.props;
+
+		return [
+			{
+				key: 'all',
+				title: translate( 'All', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'all' ),
+			},
+			{
+				key: 'engagement',
+				title: translate( 'Engagement', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'engagement' ),
+			},
+			{
+				key: 'security',
+				title: translate( 'Security', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'security' ),
+			},
+			{
+				key: 'appearance',
+				title: translate( 'Appearance', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'appearance' ),
+			},
+			{
+				key: 'writing',
+				title: translate( 'Writing', { context: 'Filter label for plugins list' } ),
+				onClick: () => this.onNavClick( 'writing' ),
+			},
+		];
+	}
+
+	getJetpackPlugins() {
+		const groupedPlugins = jetpackPlugins.map( group => this.filterGroup( group ) ).filter( g => g );
+		if ( groupedPlugins.length ) {
+			return groupedPlugins;
+		}
+
+		return <div>
+			{ this.props.translate( 'No features match your search for %s.', {
+				args: [ this.state.searchTerm ]
+			} )}
+		</div>;
+	}
+
+	filterGroup( group ) {
+		if ( 'all' === this.state.selectedGroup || group.category === this.state.selectedGroup ) {
+			const plugins = group.plugins.map( ( plugin, j ) => this.filterPlugin( plugin, j ) ).filter( p => p );
+
+			if ( plugins.length > 0 ) {
+				return <div key={ group.category }>
+					<CompactCard className="plugins-wpcom__jetpack-category-header">
+						<Gridicon icon={ group.icon } />
+						<span>
+							{ group.name }
+						</span>
+					</CompactCard>
+					{ plugins }
+				</div>;
+			}
+		}
+	}
+
+	filterPlugin( plugin, pluginKey ) {
+		if (
+			! this.state.searchTerm ||
+			-1 !== plugin.name.toLowerCase().indexOf( this.state.searchTerm.toLowerCase() )
+		) {
+			let isActive = false;
+			if (
+				'standard' === plugin.plan ||
+				( 'premium' === plugin.plan && this.props.hasPremium ) ||
+				( 'business' === plugin.plan && this.props.hasBusiness )
+			) {
+				isActive = true;
+			}
```

Comment:
this `if` is redundant and we could accomplish the same with direct assignment…

```js
const isActive = (
	'standard' === plugin.plan ||
	( … ) ||
	( … )
);
```

or

```js
const isActive = some( [
	'standard' === plugin.plan,
	'premium' === plugin.plan && this.props.hasPremium,
	'business' === plugin.plan && this.props.hasBusiness,
] );
```